framework/src/modules/chain/logic/bft.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const assert = require('assert');
+
+const __private = {
```

Comment:
prefer not to use `__private` anymore

Reply:
Why not?

Reply:
Eventually, we will move everything to typescript, and it will be easier to migrate if it just follow the standard convention. (ie: not use WeakMap)
Also, there is already private method in Node12 too, no need to hack the way to use it anymore

Reply:
In both cases that will be easier to migrate, if we clearly know which are private elements.

Reply:
it's not hard, but normal convention is easier in a way that we don't need to touch the code.
Also, it's a hack way to use the private feature which doesn't exist in node.js yet (pre-node12)

Reply:
Its a normal convention to store private data attributes for any class instance.

Reply:
Im approving this PR, but I will create meeting to discuss the usage of WeapMap.